PriceCalculatorKata/Product.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace PriceCalculatorKata
+{
+    public class Product
+    {
+        public string Name { get; set; }
+        public int UPC { get; }
+
+        private float _basePrice;
+        public float BasePrice
+        {
+            get
+            {
+                return _basePrice;
+            } set {
+                if(value > 0)
+                {
+                    _basePrice = (float)Math.Round(value, 2);
+                }   
+            }
+        }
+
+        public float PriceAfterTax
```

Comment:
nit: can be expressed as:
`        public float PriceAfterTax => (float)Math.Round(BasePrice + (TaxPercentage / 100.0f) * BasePrice,2);
`